src/driver/Firestore/IFirestore.ts

```diff
@@ -248,3 +249,56 @@ export interface IFirestoreWriteBatch {
 
     commit(): Promise<IFirestoreWriteResult[]>
 }
+
+
+export interface IFirestoreBulkWriter {
+    close(): Promise<void>
+
+    create(
+        documentRef: IFirestoreDocRef,
+        data: IFirestoreDocumentData,
+    ): Promise<IFirestoreWriteResult>
+
+    delete(
+        documentRef: IFirestoreDocRef,
+        precondition?: IPrecondition,
+    ): Promise<IFirestoreWriteResult>
+    
+    flush(): Promise<void>
+    
+    onWriteError(
+        shouldRetryCallback: (error: Error) => boolean
+    ): void
+
+    onWriteResult(
+        callback: (
+          documentRef: IFirestoreDocRef<any>,
+          result: IFirestoreWriteResult
+        ) => void
+    ): void
+
+    set(
+        documentRef: IFirestoreDocRef,
+        data: IFirestoreDocumentData,
+        options?: SetOptions,
+    ): Promise<IFirestoreWriteResult>
+
+    update(
+        documentRef: IFirestoreDocRef,
+        data: IFirestoreDocumentData,
+        precondition?: IPrecondition,
+    ): Promise<IFirestoreWriteResult>
+
+    update(
+        documentRef: IFirestoreDocRef,
+        field: string | IFieldPath,
+        value: any,
+        ...fieldsOrPrecondition: any[]
+    ): Promise<IFirestoreWriteResult>
+}
+
+export interface SetOptions {
+    merge?: boolean,
+    // Not currently implemented
+    // mergeFields?: string[],
```

Comment:
Can we include this in the interface, but throw an error if someone passes in `mergeFields`